### PR TITLE
[1.0] Backports

### PR DESCRIPTION
**Pull request by nabobalis**
backports #3680 and #3798

### CODE REVIEW

File: sunpy/database/tests/test_attrs.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def test_walker_create_vso_instrument(vso_session):
                              wavemax=7.293188082141598e-05)]
     assert len(entries) == len(expected)
     for e in entries:
-        assert e in entries
```
Comment (nabobalis) [inline sunpy/database/tests/test_attrs.py]: No wonder this test never failed.

Comment (nabobalis) [reply sunpy/database/tests/test_attrs.py]: Is this on master too?

Comment (Cadair) [reply sunpy/database/tests/test_attrs.py]: yes https://github.com/sunpy/sunpy/blob/master/sunpy/database/tests/test_attrs.py#L425

Comment (nabobalis) [reply sunpy/database/tests/test_attrs.py]: Both of them are. No wonder they didn't fail.

**Pull request merged**